src/dist/replication/lib/duplication/load_from_private_log.cpp

```diff
@@ -131,6 +151,25 @@ void load_from_private_log::replay_log_block()
                 _current->path(),
                 _start_offset);
             _counter_dup_load_file_failed_count->increment();
+            _err_file_repeats_num++;
+            if (will_fail_skip()) {
+                // skip this file
+                derror_replica("failed loading for {} times, abandon file {} and try next",
+                               _err_file_repeats_num,
+                               _current->path());
+                _err_file_repeats_num = 0;
+
+                auto prev_offset = _current_global_end_offset;
+                if (switch_to_next_log_file()) {
```

Comment:
what is going to happen when swtich fail?

Reply:
Yes. Switch fail means no next file to duplicate. So I make it retry and wait the next new file to be generated.